File: android/src/main/java/com/wogaaflutter/snowplow/SnowplowTrackerBuilder.java

package com.wogaaflutter.snowplow;

import com.snowplowanalytics.snowplow.configuration.*;
import com.snowplowanalytics.snowplow.controller.*;
import com.snowplowanalytics.snowplow.*;
import com.snowplowanalytics.snowplow.network.HttpMethod;
import com.snowplowanalytics.snowplow.tracker.LogLevel;
import com.snowplowanalytics.snowplow.util.TimeMeasure;

import android.content.Context;

import java.util.concurrent.TimeUnit;

public class SnowplowTrackerBuilder {
    public static final String HTTPS_SNOWPLOW_COLLECTOR_URL_COM = "https://snowplow.dcube.cloud";

    public static TrackerController getTracker(Context context) {
        String appId = context.getPackageName();
            
        NetworkConfiguration networkConfig = new NetworkConfiguration(HTTPS_SNOWPLOW_COLLECTOR_URL_COM, HttpMethod.POST);
        TrackerConfiguration trackerConfig = new TrackerConfiguration(appId)
                .base64encoding(true)
                .sessionContext(true)
                .platformContext(true)
                .lifecycleAutotracking(true)
                .screenViewAutotracking(true)
                .screenContext(true)
                .applicationContext(true)
                .exceptionAutotracking(true)
                .installAutotracking(true);

        SessionConfiguration sessionConfig = new SessionConfiguration(
                new TimeMeasure(30, TimeUnit.SECONDS),
                new TimeMeasure(30, TimeUnit.SECONDS)
        );

        SubjectConfiguration subjectConfiguration = new SubjectConfiguration().useragent(System.getProperty("http.agent"));

        return Snowplow.createTracker(context,
                "WogaaTracker",
                networkConfig,
                trackerConfig,
                sessionConfig,
                subjectConfiguration
        );

    }
}

File: ios/Classes/SnowplowTrackerBuilder.swift
import Foundation
import SnowplowTracker

class SnowplowTrackerBuilder: NSObject, RequestCallback {
    let kNamespace = "WogaaTracker"
    
    func getTracker(_ url: String) -> TrackerController {
        var configurations : [Configuration] = []
        
        let networkConfig = NetworkConfiguration(endpoint: url, method: .post)
        
        configurations.append(networkConfig)
        
        let trackerConfiguration = TrackerConfiguration()
            .base64Encoding(true)
            .sessionContext(true)
            .platformContext(true)
            .lifecycleAutotracking(true)
            .screenViewAutotracking(true)
            .screenContext(true)
            .applicationContext(true)
            .exceptionAutotracking(true)
            .installAutotracking(true)
        
        configurations.append(trackerConfiguration)
        
        let emitterConfiguration = EmitterConfiguration()
        emitterConfiguration.requestCallback = self
        
        configurations.append(emitterConfiguration)
    
        let sessionConfiguration = SessionConfiguration()
        if #available(iOS 10, *) {
            sessionConfiguration.foregroundTimeout = Measurement(value: 10, unit: .minutes)
            sessionConfiguration.backgroundTimeout = Measurement(value: 5, unit: .minutes)
        } else {
            // Fallback on earlier versions
        }

        configurations.append(sessionConfiguration)
        
        let subjectConfiguration = SubjectConfiguration()
        
        configurations.append(subjectConfiguration)
        
        return Snowplow.createTracker(namespace: kNamespace, network: networkConfig, configurations: configurations)
    }
        
    func onSuccess(withCount successCount: Int) {
        print("Success: \(successCount)")
    }
    
    func onFailure(withCount failureCount: Int, successCount: Int) {
        print("Failure: \(failureCount), Success: \(successCount)")
    }  
}
